### PR TITLE
clean build system, add new formatting script

### PR DESCRIPTION
**Pull request by samanpwbb**
also runs formatting script, which introduces some minor improvements to individual svg formatting

@dasulit for review

One open question: there are two ways adding ids to icons to components: with `<title>` elements or with ids on svg elements. @natslaughter or @dasulit do either of you feel _strongly_ one way or the other? Using ids is the most straightforward approach, which I implemented here, but `<title>` elements could also be used.

Moving forward, this commit shouldn't introduce any workflow changes – with the one small quality of life improvement: building layouts now happens on commit automatically whenever icons get changed.

### CODE REVIEW

**Comment by natslaughter** [comment]
@samanpwbb although I don't fully understand the pros/cons or differences between using the `id` tag or `title` tag, I will let @dasulit chime in there, but it seems that `id` suffices.

File: package.json [hunk 1]
```diff
@@ -27,5 +32,13 @@
   "bugs": {
     "url": "https://github.com/mapbox/map-icons/issues"
   },
-  "homepage": "https://github.com/mapbox/map-icons"
+  "homepage": "https://github.com/mapbox/map-icons",
```
Comment (dasulit) [inline package.json]: Ha, might as well update this homepage while we're in here

File: scripts/format-icons.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+const pReadFile = pify(fs.readFile);
+const pReaddir = pify(fs.readdir);
+
+const builder = new xml2js.Builder({ rootName: 'svg' });
+const iconPath = path.join(__dirname, '../icons');
+
+function gatherIcons() {
+  return pReaddir(iconPath).then(files => {
+    const svgFiles = files.filter(f => f.indexOf('svg') !== -1);
```
Comment (dasulit) [inline scripts/format-icons.js]: Nit, but make this more specific? `svg` => `.svg`

File: icons/zoo-15.svg [hunk 3]
```diff
@@ -1,15 +1,4 @@
...
-	l-1.7045-1.6818C12.5444,3.0772,12,2.5,11,2.5C9.5469,2.5,8.8636,3.0688,8.3879,3.8928z"/>
-</svg>
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<svg version="1.1" id="zoo-15" xmlns="http://www.w3.org/2000/svg" width="15px" height="15px" viewBox="0 0 15 15">
+  <path id="path17499" inkscape:connector-curvature="0" sodipodi:nodetypes="sssssscsccccsccscccscscccscsscccsscsss" d="&#xA;&#x9;M8.3879,3.8928C8.3879,3.8928,7.6818,5,7,5H4.5C3.8182,5,3.2016,5.4086,2.7273,5.8636L0.5,8C0.231,8.258,0.0072,8.7681,0,9&#xA;&#x9;c-0.0156,0.5,0,1,0,1s1,0,1-1V8.5l1-1h0.5l0.158,0.2892c0,0-0.9535,1.8244-0.9535,3.5289C1.7046,12,2.3864,12,2.3864,12h0.6818&#xA;&#x9;c0,0,0.3409,0,0-0.3409l-0.3409-0.3409C2.7273,10.6364,3.5002,9.6667,4,9c0,0,0.0168,1.1579,0,2c-0.008,0.4096,0.2721,1,0.6818,1&#xA;&#x9;h0.6818c0,0,0.3409,0,0-0.3409l-0.3409-0.3409C4.7105,10.7495,5.5,8.5,5.5,8.5C6.7716,8.5,7,9,8.5,9l0.3636,2.3182&#xA;&#x9;C8.975,12.0282,9.5,12,9.5,12H11c0.3409,0,0.9611-0.3115,0.3409-0.7736L11,11V9c1.2142-0.1722,2-1,2-2h1c0.3214,0,1,0,1-0.5v-1&#xA;&#x9;l-1.7045-1.6818C12.5444,3.0772,12,2.5,11,2.5C9.5469,2.5,8.8636,3.0688,8.3879,3.8928z"/>
```
Comment (dasulit) [inline icons/zoo-15.svg]: This icon, and a few others, are invalid. Looks like this inkscape key is causing problems:

screen shot 2018-11-01 at 11 19 05 am

File: icons/rail-15.svg [hunk 4]
```diff
@@ -1,49 +1,17 @@
...
-     id="path3" /></svg>
\ No newline at end of file
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<svg xmlns="http://www.w3.org/2000/svg" version="1.1" id="rail-15" width="15px" height="15px" viewBox="0 0 15 15">
+  <metadata id="metadata11">
+    <rdf:RDF>
+      <cc:Work rdf:about="">
```
Comment (dasulit) [inline icons/rail-15.svg]: This icon and some others have this error:

screen shot 2018-11-01 at 11 20 07 am

File: icons/bicycle-share-15.svg [hunk 5]
```diff
@@ -1,20 +1,7 @@
...
-	C8.8103,3.0756,8.6685,3,8.5156,2.9941H8.1445z M3,7c-1.6569,0-3,1.3432-3,3s1.3431,3,3,3s3-1.3432,3-3S4.6569,7,3,7z M12,7
-	c-1.6569,0-3,1.3432-3,3s1.3431,3,3,3s3-1.3432,3-3S13.6569,7,12,7z M3,8c1.1046,0,2,0.8954,2,2s-0.8954,2-2,2s-2-0.8954-2-2
-	S1.8954,8,3,8z M12,8c1.1046,0,2,0.8954,2,2s-0.8954,2-2,2s-2-0.8954-2-2S10.8954,8,12,8z"/>
-</svg>
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<svg version="1.1" id="bicycle-share-15" xmlns="http://www.w3.org/2000/svg" width="15px" height="15px" viewBox="0 0 15 15">
+  <sodipodi:namedview bordercolor="#666666" borderopacity="1" gridtolerance="10" guidetolerance="10" id="namedview3364" inkscape:current-layer="svg4619" inkscape:cx="0.18747857" inkscape:cy="7.9979934" inkscape:document-units="px" inkscape:object-nodes="true" inkscape:object-paths="true" inkscape:pageopacity="0" inkscape:pageshadow="2" inkscape:snap-bbox="true" inkscape:snap-intersection-paths="true" inkscape:snap-smooth-nodes="true" inkscape:window-height="755" inkscape:window-maximized="0" inkscape:window-width="1280" inkscape:window-x="0" inkscape:window-y="23" inkscape:zoom="1" objecttolerance="10" pagecolor="#ffffff" showgrid="false">
```
Comment (dasulit) [inline icons/bicycle-share-15.svg]: `bicycle-share-15` (and `bicycle-15`) is erroring here:

screen shot 2018-11-01 at 11 23 38 am

File: icons/bicycle-share-11.svg [hunk 6]
```diff
@@ -1,29 +1,7 @@
...
-	c0.6895,0.7739,1.7895,1.0434,2.7578,0.6738C10.3609,10.4667,11.0015,9.5329,11,8.4964c-0.001-1.0365-0.6439-1.9671-1.6133-2.334
-	C9.1444,6.0708,8.8928,6.0182,8.6426,6.0043L8.6426,6.0043z"/>
-</svg>
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<svg version="1.1" id="bicycle-share-11" xmlns="http://www.w3.org/2000/svg" width="11px" height="11px" viewBox="0 0 11 11">
+  <sodipodi:namedview bordercolor="#666666" borderopacity="1" gridtolerance="10" guidetolerance="10" id="namedview3424" inkscape:bbox-nodes="true" inkscape:bbox-paths="true" inkscape:current-layer="svg4764" inkscape:cx="1.2197585" inkscape:cy="4.8663967" inkscape:document-units="px" inkscape:object-nodes="true" inkscape:object-paths="true" inkscape:pageopacity="0" inkscape:pageshadow="2" inkscape:snap-bbox="true" inkscape:snap-bbox-edge-midpoints="true" inkscape:snap-intersection-paths="true" inkscape:snap-smooth-nodes="true" inkscape:window-height="784" inkscape:window-maximized="0" inkscape:window-width="1495" inkscape:window-x="16" inkscape:window-y="0" inkscape:zoom="45.254834" objecttolerance="10" pagecolor="#ffffff" showgrid="true">
```
Comment (dasulit) [inline icons/bicycle-share-11.svg]: `bicycle-share-11` is erroring here:
screen shot 2018-11-01 at 11 24 13 am

File: icons/bicycle-share-11.svg [hunk 7]
```diff
@@ -1,29 +1,7 @@
...
-	c0.584-0.221,1.2402-0.0592,1.6543,0.4082C3.8101,7.713,4.1262,7.7322,4.3329,7.549c0.2068-0.1831,0.226-0.4992,0.0428-0.706
-	C4.3755,6.8427,4.3752,6.8425,4.375,6.8422C4.0314,6.4543,3.5855,6.1936,3.1035,6.0746C2.8625,6.0151,2.6136,5.9904,2.3633,6.0043
-	L2.3633,6.0043z M8.6426,6.0043C8.3923,5.9904,8.1433,6.0151,7.9023,6.0746c-0.482,0.119-0.9298,0.3797-1.2734,0.7676
-	c-0.1834,0.2071-0.1641,0.5237,0.043,0.707s0.5237,0.1641,0.707-0.043C7.793,7.0388,8.4472,6.877,9.0312,7.098
-	C9.6153,7.319,9.9991,7.8739,10,8.4984S9.6186,9.68,9.0352,9.9027c-0.5834,0.2227-1.2389,0.062-1.6543-0.4043
-	c-0.1729-0.2153-0.4877-0.2497-0.703-0.0768s-0.2497,0.4877-0.0768,0.703c0.0106,0.0132,0.0218,0.0258,0.0337,0.0378
-	c0.6895,0.7739,1.7895,1.0434,2.7578,0.6738C10.3609,10.4667,11.0015,9.5329,11,8.4964c-0.001-1.0365-0.6439-1.9671-1.6133-2.334
-	C9.1444,6.0708,8.8928,6.0182,8.6426,6.0043L8.6426,6.0043z"/>
-</svg>
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (dasulit) [inline icons/bicycle-share-11.svg]: Do we want this `standalone` key on here?

File: README.md [hunk 8]
```diff
@@ -76,8 +76,6 @@ Every icon in Maki must pass the automated tests in [tests/maki.test.js](https:/
 - Both height and width must equal 11 or 15, and height and width must be equal.
 - Height, width, and viewbox must use pixel units.
```
Comment (dasulit) [inline README.md]: Can you document the new `id` requirement?

**Comment by samanpwbb** [comment]
oh no! Will take a closer look at these invalid svgs.

**Comment by samanpwbb** [comment]
@dasulit back to you again! Icons should now appear valid.

File: README.md [hunk 9]
```diff
@@ -11,7 +11,8 @@ This repo only contains the source SVG files. Check out [maki website](https://m
 
 - Source icons are in the SVG file format.
 - Icons are available in two sizes: 11px x 11px and 15px x 15px.
-- Icons are each a single unified SVG path, with no transforms, groups or strokes.
+- Icons should consist only of paths and groups. Paths should only have a `d` property.
+- Each Icon's svg tag should have an `id` property that corresponds to it's filename without the extension.
```
Comment (dasulit) [inline README.md]: Grammar/style recommendations:

* `Icon's` => `icon's`
* `it's filename` => `its filename`

**Pull request merged**